drone/src/cert/mod.rs

```diff
@@ -89,44 +93,61 @@ pub async fn get_certificate(
                 cluster: ClusterName::new(cluster_domain),
                 value: value.clone(),
             })
-            .await?;
+            .await
+            .context("Setting ACME DNS record")?;
+
+        if !result {
+            return Err(anyhow!("Platform rejected TXT record."));
+        }
 
         nats.publish(&SetDnsRecord {
             cluster: ClusterName::new(cluster_domain),
             kind: DnsRecordType::TXT,
             name: "_acme-challenge".to_string(),
             value,
         })
-        .await?;
-
-        if !result {
-            return Err(anyhow!("Platform rejected TXT record."));
-        }
+        .await
+        .context("Sending SetDnsRecord")?;
 
         tracing::info!("Validating challenge.");
-        let challenge = challenge.validate().await?;
-        let challenge = challenge.wait_done(Duration::from_secs(5), 3).await?;
+        let challenge = challenge.validate().await.context("Validating challenge")?;
+        let challenge = challenge
+            .wait_done(Duration::from_secs(5), 3)
+            .await
+            .context("Waiting for challenge")?;
```

Comment:
so if this errors then it'll return the context in the error?

Reply:
Yes, exactly. It's an anyhow thing.